### ISSUE TITLE
[Feature Request] Qdrant support vector delete by key word

### ISSUE THREAD

**Issue by Wendong-Fan** [camel-ai/camel#989]
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

currently we only support deleting by using id, it's hard to track, better support with deleting by key word

**Comment by koch3092** [comment]
- [ ] When initializing the Qdrant client, a global Client can be initialized to handle CRUD operations for collections.
- [ ] Filter vectors using metadata.
- [ ] Delete relevant collections by performing a fuzzy search on collection names (in conjunction with Redis).
- [ ] Check if a collection exists by performing a fuzzy search on collection names (in conjunction with Redis) and matching the full name.

For the above requirements, consider the distinction and unification between local storage and self-hosted cloud deployment.

**Comment by Wendong-Fan** [comment]
> * [ ]  When initializing the Qdrant client, a global Client can be initialized to handle CRUD operations for collections.
> * [ ]  Filter vectors using metadata.
> * [ ]  Delete relevant collections by performing a fuzzy search on collection names (in conjunction with Redis).
> * [ ]  Check if a collection exists by performing a fuzzy search on collection names (in conjunction with Redis) and matching the full name.
> 
> For the above requirements, consider the distinction and unification between local storage and self-hosted cloud deployment.

all doable

**Comment by koch3092** [comment]
thanks @Wendong-Fan , and I will take the task and develop

**Comment by koch3092** [comment]
> * [x]  When initializing the Qdrant client, a global Client can be initialized to handle CRUD operations for collections.
> * [x]  Filter vectors using metadata.
> * [ ]  Delete relevant collections by performing a fuzzy search on collection names (in conjunction with Redis).
> * [ ]  Check if a collection exists by performing a fuzzy search on collection names (in conjunction with Redis) and matching the full name.
> 
> For the above requirements, consider the distinction and unification between local storage and self-hosted cloud deployment.

I have completed the development and have a few clarifications regarding the requirements:

- Collections cannot be filtered by metadata, and it's not recommended to integrate Redis for this purpose. Qdrant tools should maintain loose coupling.
- Collections should not be deleted based solely on fuzzy search for collection types, as this approach is too risky. It’s necessary to ensure that the full collection_name is used to delete a collection.
- If there are many collections with similar names, I think the vectors should be placed in the same collection, and classification should be handled using payload.
- Self-hosted mode actually doesn't support concurrent searches. In the original design, this was addressed by managing through an in-memory list, which I believe is sufficient for local testing. In server mode, this issue doesn't need to be considered. Therefore, I believe the functionality between local storage and self-hosted Qdrant has been unified.
